core/backend/src/internal/workspace/SettingsImpl.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Workspace
+ */
+
+import * as fs from "fs-extra";
+import { parse } from "json5";
+import { extname, join } from "path";
+import { BeEvent } from "@itwin/core-bentley";
+import { LocalDirName, LocalFileName } from "@itwin/core-common";
+import { IModelJsFs } from "../../IModelJsFs";
+import { SettingName, SettingObject, SettingType, Settings } from "../../workspace/Settings";
+import { IModelHost } from "../../IModelHost";
+
+const dictionaryMatches = (d1: Settings.Dictionary.Source, d2: Settings.Dictionary.Source): boolean => {
+  return (d1.workspaceDb === d2.workspaceDb) && (d1.name === d2.name);
+};
+
+class SettingsDictionaryImpl implements Settings.Dictionary {
+  public readonly props: Settings.Dictionary.Props;
+  public readonly settings: SettingObject;
+  
+  public constructor(props: Settings.Dictionary.Props, settings: SettingObject) {
+    this.props = { ...props }; // make a copy so it can't be changed by caller
```

Comment:
I think this copy is no longer necessary since all the members are `readonly`.

Reply:
If you're concerned about caller modifying the input, `readonly` won't save you.
```ts
const props = { name: "a", priority: 1 };
settings.addDictionary(props, {});
props.name = "b";
```